### PR TITLE
Add More Retries to 400 on Query

### LINKED ISSUE #2402: monitoring.TestMonitoring.test_write_point is very flaky
https://travis-ci.org/GoogleCloudPlatform/google-cloud-python/builds/162278041#L1009

Even in non-failed builds it still takes a lot of retries:
https://travis-ci.org/GoogleCloudPlatform/google-cloud-python/builds/162244259#L995

/cc @waprin

### PR DESCRIPTION
**Pull request by waprin**
Fixes #2402.

### CODE REVIEW

**Comment by waprin** [comment]
@dhermes Can't do much about all the retrying but I think this will fix the test.

File: system_tests/monitoring.py [hunk 1]
```diff
@@ -26,6 +26,7 @@
 
 retry_404 = RetryErrors(NotFound, max_tries=5)
 retry_404_500 = RetryErrors((NotFound, InternalServerError))
+retry_400_500 = RetryErrors((BadRequest, InternalServerError))
 retry_500 = RetryErrors(InternalServerError)
 retry_503 = RetryErrors(ServiceUnavailable)
```
Comment (dhermes) [inline system_tests/monitoring.py]: This collection of retries seems to be a blunt instrument.

Can we refine them to use `error_predicate` and actually sniff around for the expected state of the error? (doesn't have to be in this PR)

**Comment by dhermes** [comment]
Thanks for the quick turnaround @waprin!

**Comment by dhermes** [comment]
> Can't do much about all the retrying but I think this will fix the test.

You can tweak the retry [defaults][1] like the initial delay and the number of retries.

[1]: https://github.com/GoogleCloudPlatform/google-cloud-python/blob/af4f346e238e1a45ff5754335b9f8a4d98708e49/system_tests/retry.py#L20-L22

**Comment by dhermes** [comment]
@waprin This doesn't fix the issue reported in #2402.

I just checked out the stacktrace and it happens when the iterator is cast to a list: https://github.com/GoogleCloudPlatform/google-cloud-python/blob/af4f346e238e1a45ff5754335b9f8a4d98708e49/system_tests/monitoring.py#L212

**Comment by waprin** [comment]
@dhermes does it look right to you now?

File: system_tests/monitoring.py [hunk 2]
```diff
@@ -206,7 +206,7 @@ def _has_timeseries(result):
                 return len(list(result)) > 0
             retry_result = RetryResult(_has_timeseries, max_tries=7)(
                 client.query)
-            return RetryErrors(BadRequest)(retry_result)
+            return RetryErrors(BadRequest, max_tries=7)(retry_result)
```
Comment (dhermes) [inline system_tests/monitoring.py]: Practice some DRY and put `7` in a variable, then use it in both places

Comment (waprin) [inline system_tests/monitoring.py]: done

**Comment by dhermes** [comment]
LGTM pending Travis.

Can you

1. Ping me once Travis goes green (I'll merge for you)
1. File an issue to make the blunt instrument sharper (i.e. blunt thing == these error retries)

**Comment by waprin** [comment]
@dhermes travis green

**Pull request merged**